src/fsharp/FSharp.Core.Unittests/FSharp.Core/Microsoft.FSharp.Collections/SeqModule.fs

```diff
@@ -302,7 +302,70 @@ type SeqModule() =
         
         CheckThrowsArgumentNullException (fun () -> Seq.choose funcInt nullSeq |> ignore) 
         ()
-    
+
+    [<Test>]
+    member this.ChunksOf() =
+
+        let verify expected actual =
+            Seq.zip expected actual
+            |> Seq.iter ((<||) VerifySeqsEqual)
+
+        // int Seq
+        verify [[1..4];[5..8]] <| Seq.chunksOf 4 {1..8}
+        verify [[1..4];[5..8];[9..10]] <| Seq.chunksOf 4 {1..10}
+
+        Seq.chunksOf 2 (Seq.initInfinite id)
+        |> Seq.take 3
+        |> verify [[0;1];[2;3];[4;5]]
+
+        Seq.chunksOf 1 (Seq.initInfinite id)
+        |> Seq.take 5
+        |> verify [[0];[1];[2];[3];[4]]
+
+        // string Seq
+        verify [["a"; "b"];["c";"d"];["e"]] <| Seq.chunksOf 2 ["a";"b";"c";"d";"e"]
+
+        // empty Seq
+        verify Seq.empty <| Seq.chunksOf 3 Seq.empty
+
+        // null Seq
+        let nullSeq:seq<_> = null
+        CheckThrowsArgumentException (fun () -> Seq.chunksOf 3 nullSeq |> ignore)
+
+        // invalidArg
+        CheckThrowsArgumentException (fun () -> Seq.chunksOf 0 {1..10} |> ignore)
+        CheckThrowsArgumentException (fun () -> Seq.chunksOf -1 {1..10} |> ignore)
+
+        ()
+
+    [<Test>]
+    member this.DivideInto() =
+
+        let verify expected actual =
+            Seq.zip expected actual
+            |> Seq.iter ((<||) VerifySeqsEqual)
+
+        // int Seq
+        Seq.divideInto 3 {1..10} |> verify (seq [ {1..4}; {5..8}; {9..10} ])
+        Seq.divideInto 3 {1..11} |> verify (seq [ {1..4}; {5..8}; {9..11} ])
+        Seq.divideInto 3 {1..12} |> verify (seq [ {1..4}; {5..8}; {9..12} ])
+
+        // string Seq
+        Seq.divideInto 3 ["a";"b";"c";"d";"e"] |> verify ([ ["a"; "b"]; ["c";"d"]; ["e"] ])
+
+        // empty Seq
+        VerifySeqsEqual [] <| Seq.divideInto 3 []
+
+        // null Seq
+        let nullSeq:seq<_> = null
+        CheckThrowsArgumentException (fun () -> Seq.divideInto 3 nullSeq |> ignore)
+
+        // invalidArg
+        CheckThrowsArgumentException (fun () -> Seq.divideInto 0 [1..10] |> ignore)
```

Comment:
This isn't done elsewhere in the tests that I can see and if done here it should probably be changed for all checks for ArgumentException